main.c

```diff
@@ -354,9 +363,26 @@ int main(void)
         switch (USB_getConnectionState())
         {
             case ST_ENUM_ACTIVE:
-                if(uartmuxState != UARTMUXSEL_MODE_MSP430)
+                uartmuxState |= UARTMUXSEL_MODE_LOCK_MSP430;
+                if((uartmuxState & 0x0F) != UARTMUXSEL_MODE_MSP430) {
```

Comment:
Why do we need the bitmask? Why not use an enum? Should it really ever have any other bits set?